### DISCUSSION TITLE
Airflow UI ignores system timezone when modifying timestamps in logs

### DISCUSSION THREAD

**Discussion by lssatvik** [apache/airflow#25842] [General]
### Apache Airflow version

2.3.3

### What happened

I changed airflow core, airflow webserver and linux system timezone to IST from UTC.
Everything is as expected except logs.
When I select UTC in UI, the log is what is in disk, IST (+5:30) time. But when I select IST in UI, the log timestamps change to UTC +5:30 +5:30.

Either the logs should contain timezone info, else we should consider the system timezone as the log timezone.

Airflow Timezone Info:-
![image](https://user-images.githubusercontent.com/3456652/185645363-887addb0-d2ba-4f5e-88af-ef7fd0b0adac.png)

System Timezone Info:-
![image](https://user-images.githubusercontent.com/3456652/185645451-8dbc4ffe-3c1b-49ff-8cdf-2e2d818ad067.png)

Log file from disk:-
![image](https://user-images.githubusercontent.com/3456652/185645599-d5003cb5-c89b-40d8-a179-37ee20831469.png)

Log file from UI with UTC time:-
![image](https://user-images.githubusercontent.com/3456652/185645849-adf10865-b5b4-40a3-b4aa-ad105e8c5f78.png)

Log file from UI with IST time:-
![image](https://user-images.githubusercontent.com/3456652/185646175-ee668aea-967c-4538-804e-b99aaa71c0d6.png)

### What you think should happen instead

UI log timestamps should be same as disk, else UI should detect what the timezone the logs were written in, else timezone info should be written into logs.

### Operating System

Ubuntu 20.04.3 LTS

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
I think it is quite expected to have the local logs in the time-zone of teh system that they run on. Changing timezone of a running system is somewhat strangea and rare event and one could say, you should also convert your current logs to the new timezone if you want to keep them in sync.

But if you really want to change the log format to contain the time-zone - you are also free to change it AIRFLOW__LOGGING__LOG_FORMAT https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#log-format

I leave it as a discussion because maybe there are some other things to consider.

**Comment by uranusjr** [comment]
Didn’t we already do this? https://github.com/apache/airflow/pull/24373
